### ISSUE TITLE
require-atomic-updates reports parameters

### ISSUE THREAD

**Issue by SyMind** [eslint/eslint#13907]
**NOTE**

This is an issue about current unit test case.

**Tell us about your environment**

* **ESLint Version:** v7.12.0
* **Node Version:** v12.16.1
* **npm Version:** v6.13.4

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "rules": {
        "require-atomic-updates": "error",
    }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
async function f(foo) {
    let buz = await get(foo.id);
    foo.bar = buz.bar;
}
```

```bash
./node_modules/.bin/eslint index.js
```

**What did you expect to happen?**

no errors

**I wonder why this case has to report an error!**

I use @babel/plugin-transform-async-to-generator plugin to transform the code, there are no variables be cached in context.

So it is never to cause race condition bugs.

```js
"use strict";

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function f(_x) {
  return _f.apply(this, arguments);
}

function _f() {
  _f = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(foo) {
    var buz;
    return regeneratorRuntime.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.next = 2;
            return get(foo.id);

          case 2:
            buz = _context.sent;
            foo.bar = buz.bar;

          case 4:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));
  return _f.apply(this, arguments);
}
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
Possible race condition: `foo.bar` might be reassigned based on an outdated value of `foo.bar`.
```

**Related**

https://github.com/eslint/eslint/issues/11723

**Are you willing to submit a pull request to fix this bug?**

Not at the moment

**Comment by SyMind** [comment]
@ljharb please

**Comment by ljharb** [comment]
@SyMind please what? I’m not an eslint collaborator.

**Comment by SyMind** [comment]
@ljharb I'm sorry.
